### ISSUE TITLE
Make queue more intuitive

### ISSUE THREAD

**Issue by sojusnik** [zyrouge/symphony#37]
Quite often you want to change the order of the songs in the queue, which is not possible at the moment. One way to elegantly solve this would be by having a way to tap and drag a song to the desired position. But the dragging should only be activated if you tap on the left side of a song, like f.i. shown on this screenshot of the awesome [AntennaPod](https://play.google.com/store/apps/details?id=de.danoeh.antennapod) app:
![grafik](https://user-images.githubusercontent.com/1245718/220199325-0478dfe9-5c92-4a8e-93fa-4a3541302361.png)

If you switch to the queue section there, you'll see that this kind of navigation is very intuitive and also allows you to scroll through a list of songs without accidentally rearranging them.

**Comment by zyrouge** [comment]
Compose doesn't yet completely support this. Got to wait till they implement this.

**Comment by zyrouge** [comment]
> https://github.com/RetroMusicPlayer/RetroMusicPlayer did it. it's also written in kotlin

It doesn't use Compose, it uses traditional XML.

**Comment by anddam** [comment]
> Compose doesn't yet completely support this. Got to wait till they implement this.

I am not into Android development so I had to search what Compose actually is, isn't [this](https://developer.android.com/jetpack/compose/touch-input/gestures#dragging) just the dragging feature?

**Comment by zyrouge** [comment]
> > Compose doesn't yet completely support this. Got to wait till they implement this.
> 
> I am not into Android development so I had to search what Compose actually is, isn't [this](https://developer.android.com/jetpack/compose/touch-input/gestures#dragging) just the dragging feature?

Dragging is supported, but it's not implementable with page transitions. Either I'll have to refactor the code or wait for this to be implemented.

**Comment by zyrouge** [comment]
Also, I don't think this is a very crucial change.

**Comment by anddam** [comment]
> Also, I don't think this is a very crucial change.

Is there a workaround to alter the order of tracks in an existing playlist?

**Comment by zyrouge** [comment]
> > Also, I don't think this is a very crucial change.
> 
> Is there a workaround to alter the order of tracks in an existing playlist?

Probably there is no real good alternative for it.

**Comment by anddam** [comment]
> Probably there is no real good alternative for it.

I cannot follow the kotlin code to figure where `Symphony.Database` is persisted, is there a way to access the playlist on the internal memory or to export and then reimport them?

The rationale is to edit them in a text editor.

**Comment by zyrouge** [comment]
> > Probably there is no real good alternative for it.
> 
> I cannot follow the kotlin code to figure where `Symphony.Database` is persisted, is there a way to access the playlist on the internal memory or to export and then reimport them?
> 
> The rationale is to edit them in a text editor.

Check `Symphony.kt` and `PlaylistRepository.kt`.

**Comment by Tom4tot** [comment]
> Also, I don't think this is a very crucial change.

I think it's actually extremely convenient and way easier to drag the current song playing around instead of deleting multiple songs. In case you ever reconsider adding this feature, I think it will be appreciated by many users.

**Comment by zyrouge** [comment]
Yes, I do understand how bad it is. If only Compose supported these, it would have been already implemented.

**Comment by TheLastGimbus** [comment]
Hi! Recently looked through all foss music players on Android, and symphony looks soooo goood 😍😍

But rearranging queue... It's a must :/// but i know how painful androi dev is... anyway, waiting for this patiently 🤞🤞

**Comment by zyrouge** [comment]
Developer reference: https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/foundation/foundation/integration-tests/foundation-demos/src/main/java/androidx/compose/foundation/demos/LazyColumnDragAndDropDemo.kt.
